### PR TITLE
Delegation Removal

### PR DESCRIPTION
**Pull request by DerTiedemann**
# Description
This PR removes delegation on the camino network.
It will no longer be possible to earn rewards by delegating stake to a validator.

# Notes
The most imporant changes touch:
- AddValidatorTx as is now cannot specifiy a delegation fee
- AdvanceTimeTx as it will now no longer need to end the delegation periods and refund the stake
- RewardValidatorTx as it will not no longer reward delegators
- chainstate in general to remove the storage of delegations
- RPC service as it will no longer accept addDelegator calls
- internal wallet which will no longer be able to issue tx that could start a delegation
- genesis, as initial staker will no longer be allowed to add a delegation fee

Everything else in this PR is either fixing function interfaces, updating test to ignore delegation or documentation

# Co-Authors
- @evlekht 
- @knikos

### CODE REVIEW

File: vms/platformvm/add_validator_tx_test.go [hunk 1]
```diff
@@ -129,36 +125,13 @@ func TestAddValidatorTxSyntacticVerify(t *testing.T) {
 		t.Fatal("should have errored because rewards owner has no addresses")
 	}
 
-	// Case: Too many shares
```
Comment (c4t-ag) [inline vms/platformvm/add_validator_tx_test.go:132]: What is the reason for removal?

File: vms/platformvm/advance_time_tx_test.go [hunk 2]
```diff
@@ -534,131 +533,8 @@ func TestWhitelistedSubnet(t *testing.T) {
...
-	onCommit, _, err = tx.UnsignedTx.(UnsignedProposalTx).Execute(vm, vm.internalState, tx)
-	assert.NoError(t, err)
-	onCommit.Apply(vm.internalState)
-	assert.NoError(t, vm.internalState.Commit())
-
-	// Test validator weight after delegation
-	vdrWeight, _ = primarySet.GetWeight(nodeID)
-	assert.Equal(t, vm.MinDelegatorStake+vm.MinValidatorStake, vdrWeight)
-}
+// ?@jax same idea here, stakeing is deeply linked with delegation removing tests that cover both...
```
Comment (c4t-ag) [inline vms/platformvm/advance_time_tx_test.go:536]: Pls. resolve and remove comments, I'm fine with removing these 2 tests...

File: vms/platformvm/reward_validator_tx_test.go [hunk 3]
```diff
@@ -195,202 +188,7 @@ func TestUnsignedRewardValidatorTxExecuteOnAbort(t *testing.T) {
...
-	newDelBalance, err := avax.GetBalance(vm.internalState, delDestSet)
-	assert.NoError(err)
-	delReward, err := math.Sub64(newDelBalance, oldDelBalance)
-	assert.NoError(err)
-	assert.Zero(delReward, "expected delegator balance not to increase")
-
-	newSupply := vm.internalState.GetCurrentSupply()
-	assert.Equal(initialSupply-expectedReward, newSupply, "should have removed un-rewarded tokens from the potential supply")
-}
+// ?@jax again not a fan
```
Comment (c4t-ag) [inline vms/platformvm/reward_validator_tx_test.go:191]: Pls. resolve / remove comment, I'm personally fine with removal, we need to add for sure deposit TestCases instead.
But they will be completely different to what we have here...

**Review by c4t-ag — CHANGES_REQUESTED** [review]
Wow, this is a huger PR as expected, but lgtm!
Pls. resolve @evlekht's comments in the code, and the one removal I commented.

@evlekht I leave it to you to agree with @DerTiedemann about the removel of the unit tests.
I'm personally fine with removal.

File: vms/platformvm/add_validator_tx_test.go (hunk 1 above)
Comment (DerTiedemann) [reply vms/platformvm/add_validator_tx_test.go:132]: The validator could set a percentage (oddly called Shares) which defined how much of a cut the validator got, when rewards were recieved. This percentage was capped, e.g. a validator could not take 100% of the potential earnings of a delegator. This was a test for that. You can cross reference this with the removal of the check in the SemanticVerify of the addValidatorTx.

Comment (c4t-ag) [reply vms/platformvm/add_validator_tx_test.go:132]: Ok, understand, even this check is "meaningless" if we start with our fixed stake, it seems for not to be really delegation related and should work in this state as expected.
@evlekht What do you think?

Comment (DerTiedemann) [reply vms/platformvm/add_validator_tx_test.go:132]: IMO this is delegation related, as it tests shares/percentages of the validator which have been removed. The case of testing, if a new validator would take too big of a cut of its delegators is not applicable anymore and hence warrants a removal.

Comment (evlekht) [reply vms/platformvm/add_validator_tx_test.go:132]: Its definitely delegation related and can be removed.
As Jax said, `Shares` its validators cut from delegators earnings. We remove delegating => we remove shares => we can remove this test

**Review by evlekht — APPROVED** [review]

**Review by c4t-ag — APPROVED** [review]

**Pull request merged**